### DISCUSSION TITLE
Demo icon component (PNG instead of SVG)

### DISCUSSION THREAD

**Discussion by ryskin** [infinitered/ignite#2298] [Q&A] [answered]
I'm curious why in dome project of ignite uses PNG instead of one file of SVG.

Why is PNG better? I suppose there was a good reason for that. I think before ignite used awesome icons.

**Comment by yulolimum** [comment, marked as answer]
HI @ryskin ! Great question!

You're right! Quick history: 
The first versions of Ignite ([2016](https://github.com/infinitered/ignite/commit/3eac1847abe4c1c328ee964d8264ddcc7fc7a548)) used icon font files bundled with the boilerplate. Later that was refactored into a [plugin](https://github.com/infinitered/ignite-vector-icons) and vector icons became opt-in. Meanwhile, the boilerplate shifted to a more leaner design (using the plugin system to augment when needed). Only icons "_required_" for the base boilerplate [were bundled ](https://github.com/infinitered/ignite-andross/tree/834777a46f561e5ce56fee976cb605adb8a4627d/boilerplate/App/Images/Icons) which were pngs. In 2018, the first version of the Icon component [was added](https://github.com/infinitered/ignite-bowser/pull/5/files) which used pngs exclusively. In 2020, [plugins became deprecated](https://shift.infinite.red/introducing-ignite-4-0-flame-1dfc891f9966) and no longer supported.

So, to summarize, the SVG icons were built-in in the very beginning and in 2.0 were extracted to a plugin and in 6.0 no longer available under the "ignite" brand.

Having said that, the svg icon plugin was just a tool to [install react-native-vector-icon](https://github.com/infinitered/ignite-vector-icons/blob/master/plugin.js#L1) which can still be achieved today albeit manually using yarn/npm. 

As to the **_why_**...

Most of the projects Infinite Red builds today are custom designs with custom icons (rather than ones obtained from icon-sets like font-awesome). The process from conception to implementation looks more/less like this:
- designer creates an icon and exports
- developer receives the icon (3 files w/ different dpis) and drops it into the icons folder
- icon is ready to use

When working with SVGs in react-native, there are a few additional steps in my experience (it might have gotten easier recently but I'm not on top of the latest). Firstly, an additional library is needed to render SVGs. Something like `react-native-vector-image` which pre-generates vector assets for both platforms (drawable xml for Android and imageset pdfs for iOS). Or,`react-native-svg` is one if you want to render the SVG elements as views. Secondly, if you're using `react-native-svg` and rendering the svg content as views, depending on how the svg was designed/exported, certain aspects of it might not work or look "not as designed". You can be stuck manually modifying the svg content until you get the desired result. Lastly, rendering SVGs is more CPU intensive than rendering bitmaps and additionally in some cases SVGs can be a larger size compared to an optimized PNG (via ImageOptim or similar). 

So you can see how things are a bit harder and more time consuming dealing with custom SVG exports - that's why we've just been using the simple-to-deal-with and proven approach. 

>Why is PNG better? I suppose there was a good reason for that. 

I think the answer to this is - it depends.. For us, using PNGs is just _simpler_ and more performant. There are a few drawbacks however. First, yes you have 3 DPIs of each icon. Additionally, you have to be aware of how you export (do the icons have padding? are the sizes the same? etc). Your icon library can definitely get out of control quickly, especially if you have a few devs on the same project.

Lastly, if you are developing something that's using FontAwesome (and the like), might as well just install the library and use the provided glyphs (no need to export icons as PNGs). Or, if you have custom icons but want to organize things in a single file, you could [generate a font](https://fontello.com/) with your custom SVGs. 

Hope this helps!

**Reply by ryskin** [reply]
Wow, I haven't expected to get so a detailed answer and that makes me feel confused to spend so much of your time )

Thank you, I got points and will use PNG in the new project. I'm really surprised to get and answer like an article, thanks again.

**Reply by yulolimum** [reply]
You're welcome - hope this helps!
